kratos/sources/kratos_application.cpp

```diff
@@ -294,5 +300,11 @@ void KratosApplication::RegisterKratosCore() {
 
     // Register ConstitutiveLaw BaseClass
     KRATOS_REGISTER_CONSTITUTIVE_LAW("ConstitutiveLaw", mConstitutiveLaw);
+
+    //Register Voxel Modeler modular components
+    RegisterVoxelMesherColoring();
+    RegisterVoxelMesherKeyPlaneGeneration();
+    RegisterVoxelMesherEntityGeneration();
+    RegisterVoxelMesherOperation();
```

Comment:
Is this really necessary? Usually this is done generic, why does the VoxelMesher require different treatment?

Reply:
I'm not sure if I understand you correctly but the only special treatment the VoxelMesher gets in terms of the registry is that it uses its own factory class. This is due to historical reasons (it started as a monolithic class and we eventually made the different components modular. The json input format that was already in use at that point is not compatible with what the standard Kratos factory expects, so we made our own factory class to parse the existing json)

Reply:
Just to clarify, this Factory is for internal use within the modeler (it is used to instantiate the requested components on demand). The modeler itself is registered using the same approach as for other C++ modelers, with the usual kratos factory class.